sale_category_discount/models/sale_order_line.py

```diff
@@ -80,8 +80,21 @@ def _recompute_price_unit(self):
                     fiscal_position=l.env.context.get(
                         'fiscal_position')
                 )
-                l.price_unit = self.env['account.tax']._fix_tax_included_price(
-                    l._get_display_price(product), product.taxes_id, l.tax_id)
+                # Consider discount policy and set correct value for
+                # price_unit.
+                if l.order_id.pricelist_id.discount_policy == \
+                        "with_discount":
+                    product_price = l._get_display_price(product)
+                else:
+                    product_price = \
+                        l.order_id.pricelist_id.get_product_price(
+                        l.product_id, l.product_uom_qty or 1.0,
+                        l.order_id.partner_id)
```

Comment:
Can above 8 lines be changed to:

```.py
                    product_price = product.with_context(
                        pricelist=self.order_id.pricelist_id.id
                    ).price
```

I guess the result should be the same, correct?